File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.py
class Solution:
    def luckyNumbers (self, matrix: List[List[int]]) -> List[int]:
        # Transpose the matrix and find the maximum in each column
        max_in_col = [max(col) for col in zip(*matrix)]

        # Find the minimum in each row
        min_in_row = [min(row) for row in matrix]

        # Find the intersection of the two lists
        result = list(set(min_in_row) & set(max_in_col))
        return result

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[]}
 */
var luckyNumbers  = function(matrix) {
  // Transpose the matrix and find the maximum in each column
  const maxInCol = matrix[0].map((_, colIndex) => Math.max(...matrix.map(row => row[colIndex])));
  console.log(maxInCol);

  // Find the minimum in each row
  const minInRow = matrix.map(row => Math.min(...row));
  console.log(minInRow);

  // Find the intersection of the two arrays
  const result = minInRow.filter(value => maxInCol.includes(value));
  return result;
};